src/OrleansAzureUtils/Hosting/AzureSilo.cs

```diff
@@ -142,15 +158,14 @@ public bool Start()
         /// <param name="deploymentId">Azure DeploymentId this silo is running under</param>
 		/// <param name="connectionString">Azure DataConnectionString. If null, defaults to the DataConnectionString setting from the Azure configuration for this role.</param>
         /// <returns><c>true</c> if the silo startup was successful</returns>
-        public bool Start(ClusterConfiguration config, string deploymentId = null, string connectionString = null)
+        internal bool Start(ClusterConfiguration config, string deploymentId, string connectionString)
         {
+            if (config != null && deploymentId != null)
+                throw new ArgumentException("Cannot use config and deploymentId on the same time");
+
             // Program ident
             Trace.TraceInformation("Starting {0} v{1}", this.GetType().FullName, RuntimeVersion.Current);
 
-            // Check if deployment id was specified
-            if (deploymentId == null)
-                deploymentId = serviceRuntimeWrapper.DeploymentId;
```

Comment:
sure, but does he have to manually pass in what was previously the default (the hosted service deployment id)?